### ISSUE TITLE
Create Internal helm-deployed Concourse Instance

### ISSUE THREAD

**Issue by scottietremendous** [concourse/concourse#2876]
**Summary**

In order to test a helm-deployed Concourse instance at a small, but realistic scale, we'd like to create a new, internal-facing Concourse deployment using the helm charts. 

For now, this deployment will exist as a type of "mock-Wings". We believe this deployment can be used for acceptance testing pipelines going forward. In order to accomplish this, we'll need to perform a few actions:

* [x] Compare all the bits from the current Wings BOSH deployment and see if we want to  keep what we currently have, or sub-in Google equivalents
* [x] Set up a Grafana dashboard via Prometheus and monitor the same SLIs as Wings
* [x] Create public address

**Comment by cirocosta** [comment]
Hey,

Some more things that @xtreme-sameer-vohra and I outlined:

- [ ] concourse: change `test:test` main password (configure github auth)
- [ ] concourse: investigate why we can't restrict log in to only concourse:Pivotal users. Every github user can log in
- [ ] concourse: setup ssl access
- [ ] generate keys for the values.yml file. No Default keys should be there

thx!

**Comment by cirocosta** [comment]
Hey,

I'm leaving the issue as "paused" now after being pretty confident that we were able to achieve a reliable Helm deployed Concourse environment.

As a result of the efforts put into it, we submitted a good deal of changes to upstream (see [charts PRs](https://github.com/helm/charts/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aopen+concourse)):

- https://github.com/helm/charts/pull/9668
- https://github.com/helm/charts/pull/9665
- https://github.com/helm/charts/pull/9614
- https://github.com/helm/charts/pull/9536
- https://github.com/helm/charts/pull/9525

Alongside those, we also have some changes that are not in the PR form given that they relate to changes that will come only after 5.x:

- [Worker rebalancing](https://github.com/cirocosta/charts/tree/concourse/worker-rebalancing)
- [Readiness and Liveness probes using Worker healthcheck endpoints](https://github.com/cirocosta/charts/tree/concourse/worker-probes)

As I see, the next step is to take that knowledge back to our internal testing under `topgun` (see https://github.com/concourse/concourse/tree/master/topgun/k8s) test suit so that we can later continuously deploy `hush-house` with full confidence - see [the topgun epic](https://github.com/concourse/concourse/issues/2742).

Please let me know if you have any remarks / questions about it.

Thanks!

**Comment by cirocosta** [comment]
Update on disk metrics:

- I tried making use of `ephemeral-storage` reservation + limits, and although that indeed works well (if you get to the limit, it'll evict your pod after some interval), but we're still unable to properly capture a decent metric from Kubernetes that would tell us the exact disk usage of such `emptyDir` (as something like xfs quotas would).

For now, I added a panel in our Grafana configuration that relates the filesystem usage of the node where the worker lands so we can have an idea of which workers are probably going to get into trouble (allowing us to alert based on that).

*ps.: if you're wondering why we make use of `emptyDir`: it's because we're then able to mount `overlayfs` - something that isn't possible if the underlying filesystem is already an overlay filesystem (see https://lkml.org/lkml/2018/1/8/81).*

**Comment by voor** [comment]
While we wait for all of these PRs to get merged into the stable chart repo, is there a helm repo or git repo we can reference to get the "best" chart?

**Comment by paulczar** [comment]
Hey folks, I'm a pivot, and also a helm charts maintainer, reach out and hit me up if you need stuff reviewed in the helm chart repo ... best way is on slack, either the kubernetes or pivotal slack.  I'm working to shepherd the above PRs through now.

**Comment by paulczar** [comment]
If y'all want to take over managing the concourse helm charts I can help with that.  Now that the chart hub supports external repos, there's no reason we couldn't move the concourse chart to its own thing. 

see https://github.com/helm/hub#distributed-search

**Comment by cirocosta** [comment]
Thanks so much for helping us out Paul!

Regarding this issue, I'll be closing it for now in favor of the issues created under https://github.com/concourse/hush-house, where the deployment lives.

Thanks for everyone who helped!
